keps/sig-apps/3329-retriable-and-non-retriable-failures/README.md

```diff
@@ -764,6 +873,13 @@ enhancement:
 
 ### Version Skew Strategy
 
+The API exposed by kube-apiserver is extended by an optional field `reason` in
```

Comment:
Your current working is not talking about version skew.

For this feature, there is a version skew when upgrading the control plane: one kube-apiserver could be in version N, while kubes-scheduler and kube-controller-manager are in N-1. https://kubernetes.io/releases/version-skew-policy/#kube-controller-manager-kube-scheduler-and-cloud-controller-manager

The strategy we are using is the feature gate. In version N, the feature gate is disabled, so nobody is writing the reason. In version N+1, the feature gate is enabled, but all the kube-apiservers at version N know how to handle the field.

Reply:
Section updated after the switch to use conditions